persistent/Database/Persist/Sql/Internal.hs

```diff
@@ -61,9 +61,13 @@ mkColumns allDefs t =
             Just (resolveTableName allDefs f, refName tn c)
         | otherwise = Nothing
     ref _ _ ("noreference":_) = Nothing
-    ref c _ (a:_)
-        | Just x <- T.stripPrefix "reference=" a =
-            Just (DBName x, refName tn c)
+    ref c fe (a:as)
+        | Just x <- T.stripPrefix "reference=" a = do
+            constraintName <- snd <$> (ref c fe as)
+            pure (DBName x, constraintName)
+        | Just x <- T.stripPrefix "constraint=" a = do
+            tableName <- fst <$> (ref c fe as)
+            pure (tableName, DBName x)
```

Comment:
Oh, I see how this is implemented now! It doesn't require a breaking change addition to a datatype, which is great. It adds a new feature to the QuasiQuoter syntax, but doesn't change the `persistent-template` library at all.

So if you have a QuasiQuoter block with `constraint=fk_name` on an old version of `persistent`, then it'll get ignored. That's okay.